modules/nf-core/seqkit/sliding/main.nf

```diff
@@ -0,0 +1,40 @@
+process SEQKIT_SLIDING {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "bioconda::seqkit=2.1.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/seqkit:2.1.0--h9ee0642_0':
+        'biocontainers/seqkit:2.1.0--h9ee0642_0' }"
+
+    input:
+    tuple val(meta), path(fastx)
+
+    output:
+    tuple val(meta), path("*.fast*"), emit: fastx
+    path "versions.yml"             , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def extension = "fastq"
+    if ("$fastx" ==~ /.+\.fasta|.+\.fa|.+\.fas|.+\.fna/) {
```

Comment:
You should probably add a `$` to the regex to make sure it matches the extension.

Alternatively, Groovy has a `.endsWith` function, so something like this should work:
```
if (fastx.endsWith('.fasta') || fastx.endsWith('.fa') [...] ) {
```

Reply:
Thanks @muffato,
I'll go with the Groovy, looks much neater than regex. That is pushed in now.

Reply:
Never mind, it's failing. I'll have a look.

Reply:
I've reverted to using the regex + end string marker. Noticed that this is potentially an issue with the other seqkit modules, where this regex is pulled from. fastq will get caught by fa.